src/openklant/config/tests/test_models.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+from django.core.cache import cache
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+
+from maykin_common.vcr import VCRTestCase
+from zgw_consumers.test.factories import ServiceFactory
+
+from openklant.components.klantinteracties.api.validators import KanaalValidator
+from openklant.components.klantinteracties.models import Klantcontact
+from openklant.config.models import ReferentielijstenConfig
+
+
+class KanaalValidatorAPITestCase(VCRTestCase):
+    service_slug = "referentielijsten-api"
+    VCR_TEST_FILES = Path("src/referentielijsten_client/tests")
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.service = ServiceFactory(
+            slug=cls.service_slug, api_root="http://localhost:8004/api/v1"
+        )
+        ReferentielijstenConfig.objects.create(
+            service=cls.service,
+            enabled=True,
+            kanalen_tabel_code="KANAAL",
+        )
+        Klantcontact.objects.create(
+            nummer="0000000001",
+            kanaal="email",
+            onderwerp="Test onderwerp",
+            inhoud="Test inhoud",
+            indicatie_contact_gelukt=True,
+            taal="nl",
+            vertrouwelijk=False,
+            plaatsgevonden_op=timezone.now(),
+        )
+
+    def setUp(self):
+        super().setUp()
+        cache.clear()
+
+    def test_validation_succeeds_for_valid_kanaal(self):
+        validator = KanaalValidator()
+        try:
+            validator("email")
+        except ValidationError:
+            self.fail("KanaalValidator raised ValidationError unexpectedly for 'email'")
+
+    def test_validation_fails_for_invalid_kanaal(self):
+        validator = KanaalValidator()
+
+        with self.assertRaises(ValidationError) as cm:
+            validator("fax")
+
+        self.assertIn("'fax' is not a valid kanaal", str(cm.exception))
+        self.assertIn("email", str(cm.exception))
+        self.assertIn("phone", str(cm.exception))
```

Comment:
Instead of testing this by calling the validator directly, could you test this by making API calls (POST/PUT/PATCH) to `/klantcontacten`?

I think it would also be good to have a test for the behavior of the API if:
1. referentielijsten returns a 400 or 500 error
2. a requestexception or timeout occurs (you could use https://maykin-django-common.readthedocs.io/en/latest/reference/vcr.html#maykin_common.vcr.VCRMixin.vcr_raises for this)